com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/DotsDeformation.hlsl

```diff
@@ -0,0 +1,67 @@
+#ifndef SKINNING_INCLUDED
+#define SKINNING_INCLUDED
+
+#if defined(DOTS_INSTANCING_ON)
+struct DeformedVertexData
+{
+    float3 Position;
+    float3 Normal;
+    float3 Tangent;
+};
+
+uniform StructuredBuffer<DeformedVertexData> _DeformedMeshData;
+uniform StructuredBuffer<DeformedVertexData> _PreviousFrameDeformedMeshData;
+
+void DOTS_Deformation(inout AttributesMesh input)
+{
+    // x = curr frame index
+    // y = prev frame index
+    // z = deformation check (0 = no deformation, 1 = has deformation)
+    // w = skinned motion vectors
+    const int4 deformProperty = asint(unity_ComputeMeshIndex);
+    const int doSkinning = deformProperty.z;
+    if (doSkinning == 1)
+    {
+        const int streamIndex = _HybridDeformedVertexStreamIndex;
+        const int startIndex = deformProperty[streamIndex];
```

Comment:
other note: we should still return current position even if skinned motion vectors are disabled? Otherwise motion vec will be written to wrong position (for instance camrea motion vec)

Reply:
Fixed issue with skinning for vertices that are not using skinned motion vectors. Won't resolve conversation though because of your comment.